### PR TITLE
autogenerate indices using parallel loader

### LINKED ISSUE #136: Ingesting CSV data with a `unique_` parameter that is not indexed is slow
It is easy to overlook creating an index for CSV data when ingesting for the first time.  We shoudl at least warn the user when runnign an ingestion script using conditional add on a property that is not indexed that it is going to be slow (particularly when the objects already exist)

### PR DESCRIPTION
**Pull request by JoshStoddard**
When using data loaders, it will attempt to create an index for any property we use for conditional add (ie. with `constraint_` prefix), if it is not already indexed.

Fixes #136

### CODE REVIEW

File: aperturedb/BBoxDataCSV.py [hunk 1]
```diff
@@ -60,14 +60,21 @@ class BBoxDataCSV(CSVParser.CSVParser):
     def __init__(self, filename, df=None, use_dask=False):
 
         super().__init__(filename, df=df, use_dask=use_dask)
-        if not use_dask:
```
Comment (gsaluja9) [inline aperturedb/BBoxDataCSV.py:63]: This check ensures that a CSV file is not read at the first pass when dask mode is used. It's a performance optimization, but it's impact can be observed when using a file with millions of rows. I can explain more offline.

**Review by gsaluja9 — CHANGES_REQUESTED** [review]

File: aperturedb/BBoxDataCSV.py (hunk 1 above)
Comment (JoshStoddard) [reply aperturedb/BBoxDataCSV.py:63]: Thanks, sorry, for got to hit send 🤦 

My interpretation was that this was only making use of the header, which should already be populated by `super().__init__()` and `validation()`. 

In any event, for this PT, I only need the header row.  Is there a way to get that without having to read the whole file?

Comment (gsaluja9) [reply aperturedb/BBoxDataCSV.py:63]: I see. I think we can merge it and worry about the optimization part later.

**Review by gsaluja9 — APPROVED** [review]

File: aperturedb/ParallelLoader.py [hunk 2]
```diff
@@ -16,6 +16,59 @@ def __init__(self, db, dry_run=False):
...
+
+        if len(new_indices) == 0:
+            return
+
+        logger.info(
+            f"Creating {len(new_indices)} indices: {new_indices}.")
+
+        create_indices = [{"CreateIndex": idx} for idx in new_indices]
+
+        res, _ = self.db.query(
```
Comment (luisremis) [inline aperturedb/ParallelLoader.py:60]: same here, we should just use [this method on Utils.py](https://github.com/aperture-data/aperturedb-python/blob/develop/aperturedb/Utils.py#L207).

File: aperturedb/ParallelLoader.py [hunk 3]
```diff
@@ -16,6 +16,59 @@ def __init__(self, db, dry_run=False):
         super().__init__(db, dry_run=dry_run)
         self.type = "element"
 
+    def get_existing_indices(self):
```
[the remark this answers is not in the document]
Comment (luisremis) [reply aperturedb/ParallelLoader.py:19]: I will create an issue for re-factoring this later, it is not a big deal.

**Pull request merged**